File: MercadoPagoSDKExamplesObjectiveC/MercadoPagoSDKExamplesObjectiveC/Hooks/SecondHookViewController.m
//
//  SecondHookViewController.m
//  MercadoPagoSDKExamplesObjectiveC
//
//  Created by Eden Torres on 11/27/17.
//  Copyright © 2017 MercadoPago. All rights reserved.
//

#import "SecondHookViewController.h"

@interface SecondHookViewController ()
@property (weak, nonatomic) IBOutlet UILabel *paymentMethodLabel;
@property PaymentData *paymentData;

@end

@implementation SecondHookViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"SecondHookViewController loaded");
}

- (IBAction)didTapOnNext {
    if (self.actionHandler != nil) {
        [self.actionHandler nextWithHook:[self getStep]];
    }
}

- (UIView * _Nonnull)render {
    return self.view;
}

- (void)didReciveWithHookStore:(HookStore * _Nonnull)hookStore {
    self.paymentData = [hookStore getPaymentData];
}

- (enum HookStep)getStep {
    return HookStepSTEP2;
}

- (void)renderDidFinish {
    self.paymentMethodLabel.text = self.paymentData.paymentMethod._id;
    NSLog(@"renderDidFinish");
}

- (BOOL)shouldShowBackArrow {
    return YES;
}

- (BOOL)shouldShowNavigationBar {
    return YES;
}

- (NSString * _Nullable)titleForNavigationBar {
    return @"Soy hook 2";
}

- (UIColor * _Nullable)colorForNavigationBar {
    return nil;
}
@end

File: PodTester/PodTester/Hooks/SecondHookViewController.swift
//
//  SecondHookViewController.swift
//  PodTester
//
//  Created by Juan sebastian Sanzone on 4/12/17.
//  Copyright © 2017 AUGUSTO COLLERONE ALFONSO. All rights reserved.
//

import UIKit
import MercadoPagoSDK

class SecondHookViewController: UIViewController {
    
    var actionHandler: PXActionHandler?
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    @IBAction func didTapOnNext() {
        actionHandler?.next()
    }
}

//MARK: - Hooks implementation.
extension SecondHookViewController: PXHookComponent {
    
    func hookForStep() -> PXHookStep {
        return .AFTER_PAYMENT_METHOD_CONFIG
    }
    
    func render() -> UIView {
        return self.view
    }
    
    func titleForNavigationBar() -> String? {
        return "Hook 2"
    }
}


//MARK: - Presentation-Navigation
extension SecondHookViewController {
    static func get() -> SecondHookViewController {
        return HooksNavigationManager().getSecondHook()
    }
}
